### PR TITLE
misc. say fixes (mostly custom say emote fixes)

### PR DESCRIPTION
**Pull request by tontyGH**
## About The Pull Request

- dead_say no longer checks message modifiers for no reason (you can now :trollface: at the beginning of messages)
- custom say emotes now actually make you emote instead of a weird say message
- custom say emotes now actually punctuate over the radio
- putting a space between the separator and the emote doesn't leave a gap anymore
- (moved the punctuation part of treat_message_min to it's own proc)

## Why It's Good For The Game

the custom say stuff is really weird and jank

## Testing Photographs and Procedure
<details>
<summary>Screenshots&Videos</summary>
-- Proper emoting and punctuation --

https://user-images.githubusercontent.com/39193182/223626072-1835f580-3021-43f2-abca-1709fc551eae.mp4

-- Trollface --

https://user-images.githubusercontent.com/39193182/223626247-43814f98-d059-4d7b-8410-630f6410b94d.mp4

</details>

## Changelog
:cl: tonty
fix: you can finally emote at the beginning of messages in dchat
fix: custom say emotes properly punctuate over the radio
fix: custom say emotes actually EMOTE NOW (can you believe it)
fix: you can have spaces around the custom say separator without it looking jank
code: text.dm has a punctuate proc now
/:cl:

### CODE REVIEW

**Comment by Chirynnn** [comment]
> custom say emotes now actually punctuate over the radio
Please, no. Punctuation already ruins regular say messages...

**Comment by tontyGH** [comment]
> Please, no. Punctuation already ruins regular say messages...

Having emotes over the radio be the _only_ messages that don't punctuate looks super jank

**Review by itsmeow — APPROVED** [review]
Very nice.

**Comment by itsmeow** [comment]
> mobs have a punctuate proc now

your CL is technically wrong, it's a global proc not a mob proc. it should probably go in text helpers or something but it's fine where it is I guess

**Comment by tontyGH** [comment]
@itsmeow i think my soul would've left my body if this was merged without me noticing that thank you so much

**Comment by tontyGH** [comment]
Oh wait I thought the workflow had to be approved again for it to be merged, my bad

File: code/_globalvars/misc.dm [hunk 1]
```diff
@@ -49,8 +49,8 @@ GLOBAL_PROTECT(topic_servers)
 GLOBAL_LIST_EMPTY(tooltips)
 
 //Should be in the form of "tag to be replaced" = list("replacement for beginning", "replacement for end")
-GLOBAL_LIST_INIT(markup_tags, list("/"  = list("<i>", "</i>"),
+GLOBAL_LIST_INIT(markup_tags, list("_"  = list("<i>", "</i>"),
 								   "**" = list("<b>", "</b>")))
 //Should be in the form of "((\\W|^)@)(\[^@\]*)(@(\\W|$)), "g"", where @ is the appropriate tag from markup_tags
-GLOBAL_LIST_INIT(markup_regex, list("/"  = new /regex("((\\W|^)_)(\[^_\]*)(_(\\W|$))", "g"),
+GLOBAL_LIST_INIT(markup_regex, list("_"  = new /regex("((\\W|^)_)(\[^_\]*)(_(\\W|$))", "g"),
```
Comment (itsmeow) [inline code/_globalvars/misc.dm:55]: what's going on here? why is this change here?

Comment (tontyGH) [reply code/_globalvars/misc.dm:55]: this was back when the italics tag was "/", when the original guy changed the tag to "_" he forgot to change the keys to reflect that

this doesn't change or break anything if that's the concern it's just for clarity's sake

Comment (tontyGH) [reply code/_globalvars/misc.dm:55]: the comment above it also says 
> //Should be in the form of "tag to be replaced" = list("replacement for beginning", "replacement for end")

so fixing this respects how it's supposed to be set up

Comment (itsmeow) [reply code/_globalvars/misc.dm:55]: How does this not fix or break anything? Is the code unused?

Comment (tontyGH) [reply code/_globalvars/misc.dm:55]: it's used, it's just that the keys themselves aren't ever directly read, they're just for indexing the markup lists
image

all I did was change the key strings to be the actual currently used italics tag for clarity, it won't affect anything

Comment (tontyGH) [reply code/_globalvars/misc.dm:55]: i'm really bad at explaining things

the keys are just names meant to show what the markup tag is, I just changed the italics ones to what the tag is supposed to be, which is "_"

**Review by itsmeow — APPROVED** [review]
